mne/io/nedf/nedf.py

```diff
@@ -0,0 +1,181 @@
+
+    dt = []  # dtype for the binary data block
+    datadt = []  # dtype for a single EEG sample
+    with open(filename, 'rb') as f:
+        header = f.read(10240)
+
+    headerend = header.find(b'\0')
+    if headerend == -1:
+        raise RuntimeError('End of header null not found')
+    headerxml = ElementTree.fromstring(header[:headerend])
+    nedfversion = headerxml.findtext('NEDFversion', '')
+    if nedfversion not in ['1.3', '1.4']:
+        warn('Unexpected NEDFversion, hope this works anyway')
+    if headerxml.findtext('AdditionalChannelStatus', 'OFF') != 'OFF':
+        raise RuntimeError('Unknown additional channel, aborting.')
+
+    n_acc = int(headerxml.findtext('NumberOfChannelsOfAccelerometer', 0))
+    if n_acc:
+        # expect one sample of u16 accelerometer data per block
+        dt.append(('acc', '>u2', (n_acc,)))
+
+    eegset = headerxml.find('EEGSettings')
+    if eegset is None:
+        raise RuntimeError('No EEG channels found')
+    nchantotal = int(eegset.find('TotalNumberOfChannels').text)
+    info['nchan'] = nchantotal
+    info['sfreq'] = int(eegset.find('EEGSamplingRate').text)
+    info['ch_names'] = [e.text for e in eegset.find('EEGMontage')]
+    # expect nchantotal uint24s
+    datadt.append(('eeg', 'B', (nchantotal, 3)))
+
+    info['units'] = eegset.find('EEGUnits')
+
+    if headerxml.find('STIMSettings'):
+        # 2* -> two stim samples per eeg sample
+        datadt.append(('stim', 'B', (2, nchantotal, 3)))
+
+    if 'AdditionalChannelStatus' in headerxml:
+        raise RuntimeError('Unexpected AdditionalChannelStatus')
+
+    if headerxml.findtext('stepDetails/DeviceClass', '') == 'STARSTIM':
+        warn('Found Starstim, not sure how to handle this')
+
+    # Trigger data: 4 bytes in newer versions, 1 byte in older versions
+    trigger_type = '>i4' if headerxml.findtext('NEDFversion') else 'B'
+    datadt.append(('trig', trigger_type))
+    # 5 data samples per block
+    dt.append(('data', np.dtype(datadt), (5,)))
+
+    date = headerxml.findtext('StepDetails/StartDate_firstEEGTimestamp', '')
+    info['meas_date'] = datetime.fromtimestamp(int(date) / 1000, timezone.utc)
+    return info, np.dtype(dt)
+
+
+def _read_nedf_eeg(filename: str):
+    """Read header info and EEG data from an .nedf file.
+
+    Parameters
+    ----------
+    filename : str
+        Path to the .nedf file.
+
+    Returns
+    -------
+    eeg : array, shape (n_samples, n_channels)
+        Unscaled EEG data.
+    info : dict
+        Information from the file header.
+    triggers : array, shape (n_annots, 2)
+        Start samples and values of each trigger.
+    scale : float
+        Scaling factor for the EEG data.
+    """
+    info, dt = parse_nedf_header(filename)
+
+    # to quote the NEDF EEGLAB plugin
+    # (https://www.neuroelectrics.com/resources/software/):
+    # "binary data will always start at byte 5120"
+    # newer versions have 10kB for the header
```

Comment:
OK, but why is "binary data will always start at byte 5120" relevant if the code only uses 10kB? If 5120 is never used, it's better to remove this statement to avoid confusion.

Reply:
Agreed, it's removed in a new commit. I'll push it with the unit tests so the CI doesn't feel the need to test everything again.